common/sgx/collateral.h

```diff
@@ -65,10 +68,47 @@ oe_result_t oe_get_sgx_quote_verification_collateral(
  * oe_get_sgx_quote_verification_collateral_from_certs().
  *
  * @param[in] args The quote verification collateral.
+ * @deprecated
  */
 void oe_free_sgx_quote_verification_collateral_args(
     oe_get_sgx_quote_verification_collateral_args_t* args);
 
+/**
+ * Fetch quote verification collateral from the quote provider given the PCK
+ * certificate, CA certificate and custom parameters for third party collateral
+ * caching service.
+ *
+ * Caller is responsbile for freeing the quote verification collateral resources
+ * by calling oe_free_sgx_quote_verification_collateral_with_params_args().
+ *
+ * @param[in] leaf_cert The PCK certificate.
+ * @param[out] args Quote verification collateral.
+ */
+oe_result_t oe_get_sgx_quote_verification_collateral_with_params_from_certs(
+    oe_cert_t* leaf_cert,
+    oe_get_sgx_quote_verification_collateral_with_params_args_t* args);
+
+/**
+ * Get the quote verification collateral from the quote provider with additional
+ * OE SDK Application specified parameters for third party cloud collateral
+ * caching service. Caller is responsible for configuring the quote verification
+ * collateral input parameters.
+ *
+ * @param[in,out] args The quote verification collateral.
+ */
+oe_result_t oe_get_sgx_quote_verification_collateral_with_params(
+    oe_get_sgx_quote_verification_collateral_with_params_args_t* args);
+
+/**
+ * Free resources allocated by
+ * oe_get_sgx_quote_verification_collateral_with_params() and
```

Comment:
91-95 do not mention any need to free resources, nor does line 71 point to this API.